calyx/opt/src/passes_experimental/metadata_table_gen.rs

```diff
@@ -55,57 +75,168 @@ impl Named for Metadata {
     }
 }
 
+impl ConstructVisitor for Metadata {
+    fn from(_ctx: &calyx_ir::Context) -> calyx_utils::CalyxResult<Self>
+    where
+        Self: Sized,
+    {
+        Ok(Self::new())
+    }
```

Comment:
This is fine for the initial constructor but we should also implement
`start_context` and take the source info from the context if it exists, only
using a fresh source info map when there is not one present. Otherwise this pass
will end up trampling pre-existing metadata info